durabletask/worker.py

```diff
@@ -606,17 +671,31 @@ def process_event(self, ctx: _RuntimeOrchestrationContext, event: pb.HistoryEven
                         self._logger.warning(
                             f"{ctx.instance_id}: Ignoring unexpected subOrchestrationInstanceFailed event with ID = {task_id}.")
                     return
-                sub_orch_task.fail(
-                    f"Sub-orchestration task #{task_id} failed: {failedEvent.failureDetails.errorMessage}",
-                    failedEvent.failureDetails)
-                ctx.resume()
+                if isinstance(sub_orch_task, task.RetryableTask):
+                    if sub_orch_task._retry_policy is not None:
+                        next_delay = sub_orch_task.compute_next_delay()
+                        if next_delay == None:
+                            sub_orch_task.fail(
+                                f"Sub-orchestration task #{task_id} failed: {failedEvent.failureDetails.errorMessage}",
```

Comment:
One potential improvement we could add here is including the number of retries in this error message.

Reply:
This is logged when it finally fails after Either 1. All retries have exhausted Or 2. Retry timed out.
This is not actually logged with each retry. 

I can add such logging at each attempt count but would need some new definition to have this understanding that what attempt number is it And, if retry has timed out, etc.
Do you think that is required at each retry/attempt?

Reply:
I'm not suggesting we log on each retry (though we could consider this too). I'm simply suggesting that the error message could include the number of retries that were attempted. For example,
```suggestion
                                f"Sub-orchestration task #{task_id} failed after {attempt_count} attempt(s): {failedEvent.failureDetails.errorMessage}",
```